src/lib/insights/listener.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'preact-compat';
+import PropTypes from 'prop-types';
```

Comment:
We don't need `PropTypes` anymore in InstantSearch.js since it's a TypeScript component.

Reply:
I think we can keep both.
If you look at other codebases using typescript (including [react-instantsearch](https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-dom/src/components/Hits.tsx#L2)), that's what they seem to do. https://github.com/apollographql/react-apollo/blob/master/src/ApolloConsumer.tsx#L2

Reply:
These projects are for React consumers. In InstantSearch.js, this is only internal. This is therefore redundant.

Reply:
I still think they are helpful as they run at a different time and catch property type mutations for example. 
I've removed it here: b8e063bd